runtime/devnet/src/extensions/mod.rs

```diff
@@ -373,118 +285,27 @@ impl TryFrom<u8> for FuncId {
-		// NftsKeys::CustomAttribute(account, collection, item, key) => {
-		// 	env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-		// 	Ok(pallet_nfts::Pallet::<T>::custom_attribute(&account, &collection, &item, &key)
-		// 		.encode())
-		// },
-		NftsKeys::SystemAttribute(collection, item, key) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::system_attribute(&collection, item.as_ref(), &key)
-				.encode())
-		},
-		NftsKeys::CollectionAttribute(collection, key) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_nfts::Pallet::<T>::collection_attribute(&collection, &key).encode())
-		},
-	}
-}
-
-fn read_assets_state<T, E>(
-	key: AssetsKeys,
+fn read_fungibles_state<T, E>(
+	key: Read<T>,
 	env: &mut Environment<E, BufInBufOutState>,
 ) -> Result<Vec<u8>, DispatchError>
 where
 	T: pallet_contracts::Config
-		+ pallet_assets::Config<TrustBackedAssetsInstance, AssetId = AssetId>,
+		+ pallet_assets::Config<TrustBackedAssetsInstance, AssetId = AssetId>
+		+ fungibles::Config,
 	E: Ext<T = T>,
 	T: frame_system::Config<AccountId = sp_runtime::AccountId32>,
 {
+	env.charge_weight(T::DbWeight::get().reads(1_u64))?;
 	match key {
-		TotalSupply(id) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_assets::Pallet::<T, TrustBackedAssetsInstance>::total_supply(id).encode())
-		},
-		BalanceOf(id, owner) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_assets::Pallet::<T, TrustBackedAssetsInstance>::balance(id, &owner.0.into())
-				.encode())
-		},
+		TotalSupply(id) => Ok(fungibles::Pallet::<T>::total_supply(id).encode()),
+		BalanceOf(id, owner) => Ok(fungibles::Pallet::<T>::balance_of(id, &owner).encode()),
 		Allowance(id, owner, spender) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(pallet_assets::Pallet::<T, TrustBackedAssetsInstance>::allowance(
-				id,
-				&owner.0.into(),
-				&spender.0.into(),
-			)
-			.encode())
-		},
-		TokenName(id) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(<pallet_assets::Pallet<T, TrustBackedAssetsInstance> as MetadataInspect<
-				AccountId,
-			>>::name(id)
-			.encode())
-		},
-		TokenSymbol(id) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(<pallet_assets::Pallet<T, TrustBackedAssetsInstance> as MetadataInspect<
-				AccountId,
-			>>::symbol(id)
-			.encode())
-		},
-		TokenDecimals(id) => {
-			env.charge_weight(T::DbWeight::get().reads(1_u64))?;
-			Ok(<pallet_assets::Pallet<T, TrustBackedAssetsInstance> as MetadataInspect<
-				AccountId,
-			>>::decimals(id)
-			.encode())
+			Ok(fungibles::Pallet::<T>::allowance(id, &owner, &spender).encode())
 		},
+		TokenName(id) => Ok(fungibles::Pallet::<T>::token_name(id).encode()),
+		TokenSymbol(id) => Ok(fungibles::Pallet::<T>::token_symbol(id).encode()),
+		TokenDecimals(id) => Ok(fungibles::Pallet::<T>::token_decimals(id).encode()),
 		// AssetsKeys::AssetExists(id) => {
```

Comment:
Maybe add a TODO on this commented code?

Reply:
Removed